wheatley/tower.py

```diff
@@ -92,56 +93,56 @@ def ring_bell(self, bell: Bell, is_handstroke: bool):
             self.logger.error(e)
             return False
 
-    def is_bell_assigned_to(self, bell: Bell, user_name: Optional[str]):
+    def is_bell_assigned_to(self, bell: Bell, user_name: Optional[str]) -> bool:
         """ Returns true if a given bell is assigned to the given user name. """
         try:
             assigned_user_id = self._assigned_users.get(bell, None)
-            if assigned_user_id is None and user_name is None:
-                return True
+            if assigned_user_id is None:
+                return user_name is None
             return self.user_name_from_id(assigned_user_id) == user_name
         except KeyError:
             return False
 
-    def user_name_from_id(self, user_id: int) -> str:
+    def user_name_from_id(self, user_id: int) -> Optional[str]:
         """
         Converts a numerical user ID into the corresponding user name, returning None if user_id is not in
         the tower.
         """
         return self._user_name_map.get(user_id)
 
-    def get_stroke(self, bell: Bell):
+    def get_stroke(self, bell: Bell) -> Optional[bool]:
```

Comment:
Is it perhaps worth introducing a type for Stroke?
```python
Stroke = NewType('Stroke', bool)
```

Reply:
Ah I _thought_ I'd forgotten something...

Reply:
So I've made a specific class for `Stroke` - I wanted to add things like overriding the display functions and making methods `is_hand()` and `is_back()` to help the code be more explicit.  But this has now been done.